### PR TITLE
Drone: use nightly build cuda docker images

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Update used docker images to the one we regularly update...
Next PR: switch docker image to PT 1.6 version fro Drone testing

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## After PR merged

- [ ] remove old images from Docker Hub
https://hub.docker.com/repository/docker/pytorchlightning/pytorch_lightning/tags?page=1&name=cuda-extras

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
failing on docker using:
```

ImportError while loading conftest '/drone/src/tests/conftest.py'.
--
226 | tests/__init__.py:13: in <module>
227 | os.mkdir(TEMP_PATH)
228 | E   PermissionError: [Errno 13] Permission denied: '/drone/src/test_temp'
229 | Traceback (most recent call last):
230 | File "/home/flash/miniconda/envs/lightning/lib/python3.7/site-packages/coverage/cmdline.py", line 740, in do_run
231 | runner.run()
232 | File "/home/flash/miniconda/envs/lightning/lib/python3.7/site-packages/coverage/execfile.py", line 247, in run
233 | exec(code, main_mod.__dict__)
234 | File "/home/flash/miniconda/envs/lightning/lib/python3.7/site-packages/py/test.py", line 4, in <module>
235 | sys.exit(pytest.main())
236 | SystemExit: ExitCode.USAGE_ERROR
```

**Comment by Borda** [comment]
need to rebuild the docker image in #3661

**Comment by ydcjeff** [comment]
Could this error be the non-root user creation?

**Comment by Borda** [comment]
> Could this error be the non-root user creation?

it seems so... I ll tttry to get a version with sudo like we have for XLA...

**Comment by Borda** [comment]
@ydcjeff I have built this one locally and pushed it to Docker hub directly to test that it fine to run the tests...

**Comment by Borda** [comment]
seems we have failing test for PR 1.6
```
FAILED tests/models/test_gpu.py::test_multi_gpu_model_ddp[variation_fit---max_epochs 1 --gpus 2 --distributed_backend ddp]
FAILED tests/models/test_gpu.py::test_multi_gpu_model_ddp[variation_test---max_epochs 1 --gpus 2 --distributed_backend ddp]
```
is it something you have seen? @awaelchli

**Comment by Borda** [comment]
probably the DD just needs to set some env variables...

File: pytorch_lightning/accelerators/ddp_accelerator.py [hunk 1]
```diff
@@ -144,8 +144,9 @@ def train(self):
         model = self.trainer.model
 
         results = self.ddp_train(process_idx=self.task_idx, model=model)
-        if 'WORLD_SIZE' in os.environ:
-            del os.environ['WORLD_SIZE']
+        # TODO: try this skip
+        # if 'WORLD_SIZE' in os.environ:
+        #     del os.environ['WORLD_SIZE']
```
Comment (Borda) [inline pytorch_lightning/accelerators/ddp_accelerator.py:149]: @awaelchli is seems there is no restore anywhere else... so how is it possible it passes on master now?

Comment (Borda) [reply pytorch_lightning/accelerators/ddp_accelerator.py:155]: any idea why the env deletion was added here? especially after training...

**Comment by Borda** [comment]
when I run the tests on similar machine...
```
=================================================================== short test summary info =====================================================================
FAILED tests/backends/test_ddp.py::test_multi_gpu_model_ddp_fit_only[--max_epochs 1 --gpus 2 --distributed_backend ddp] - FileNotFoundError: [Errno 2] No such ...
FAILED tests/backends/test_ddp.py::test_multi_gpu_model_ddp_test_only[--max_epochs 1 --gpus 2 --distributed_backend ddp] - FileNotFoundError: [Errno 2] No such...
FAILED tests/backends/test_ddp.py::test_multi_gpu_model_ddp_fit_test[--max_epochs 1 --gpus 2 --distributed_backend ddp] - FileNotFoundError: [Errno 2] No such ...
FAILED tests/models/test_amp.py::test_amp_with_apex - RuntimeError: CUDA error: no kernel image is available for execution on the device
FAILED tests/trainer/test_trainer.py::test_gradient_clipping_fp16 - AssertionError: Gradient norm != 1.0: nan
FAILED tests/trainer/optimization/test_manual_optimization.py::test_multiple_optimizers_manual_native_amp - RuntimeError: unscale_() has already been called on...
FAILED tests/trainer/optimization/test_manual_optimization.py::test_multiple_optimizers_manual_apex - RuntimeError: CUDA error: no kernel image is available fo...
======================================== 7 failed, 1351 passed, 19 skipped, 1 xfailed, 2544 warnings in 603.30s (0:10:03) ========================================

```

**Comment by Borda** [comment]
Ok, removing Conda for multiple issues...
Actually failing Apex install for https://github.com/NVIDIA/apex/issues/955#issuecomment-716047753

**Comment by Borda** [comment]
For switch to 1.6 we have one failing test `FAILED tests/models/test_horovod.py::test_horovod_amp - assert 1 == 0`

**Review by awaelchli — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Great Addition :)

**Pull request merged**